src/main/java/org/candlepin/model/CloudIdentifierFacts.java

```diff
@@ -0,0 +1,152 @@
+     */
+    public static String extractCloudAccountId(Map<String, String> facts) {
+        if (facts.containsKey(AWS_ACCOUNT_ID.getValue())) {
+            return facts.get(AWS_ACCOUNT_ID.getValue());
+        }
+        if (facts.containsKey(AZURE_SUBSCRIPTION_ID.getValue())) {
+            return facts.get(AZURE_SUBSCRIPTION_ID.getValue());
+        }
+        if (facts.containsKey(GCP_PROJECT_ID.getValue())) {
+            return facts.get(GCP_PROJECT_ID.getValue());
+        }
+        return null;
+    }
+
+    /**
+     * Extracts the cloud instance ID from the provided map of facts.
+     *
+     * @param facts
+     *      a map containing various key-value pairs representing facts about the consumer
+     *
+     * @return the instance ID of the cloud instance if found, otherwise null
+     */
+    public static String extractCloudInstanceId(Map<String, String> facts) {
+        if (facts.containsKey(AWS_INSTANCE_ID.getValue())) {
+            return facts.get(AWS_INSTANCE_ID.getValue());
+        }
+        if (facts.containsKey(AZURE_INSTANCE_ID.getValue())) {
+            return facts.get(AZURE_INSTANCE_ID.getValue());
+        }
+        if (facts.containsKey(GCP_INSTANCE_ID.getValue())) {
+            return facts.get(GCP_INSTANCE_ID.getValue());
+        }
+        return null;
+    }
+
+    /**
+     * Determines the cloud provider (AWS, Azure, or GCP) based on the provided map of facts.
+     *
+     * @param facts
+     *      a map containing various key-value pairs representing facts about the consumer
+     *
+     * @return a short name representing the detected cloud provider ("AWS", "AZURE", or "GCP"),
+     *         or null if no cloud provider facts are found
+     *
+     * @throws IllegalStateException if multiple cloud provider facts are detected
+     */
+    public static String extractCloudProviderShortName(Map<String, String> facts) {
+        boolean hasAWS = facts.containsKey(AWS_ACCOUNT_ID.getValue()) ||
+            facts.containsKey(AWS_INSTANCE_ID.getValue()) ||
+            facts.containsKey(AWS_MARKETPLACE_PRODUCT_CODES.getValue()) ||
+            facts.containsKey(AWS_BILLING_PRODUCTS.getValue());
+        boolean hasAzure = facts.containsKey(AZURE_SUBSCRIPTION_ID.getValue()) ||
+            facts.containsKey(AZURE_INSTANCE_ID.getValue()) || facts.containsKey(AZURE_OFFER.getValue());
+        boolean hasGCP = facts.containsKey(GCP_PROJECT_ID.getValue()) ||
+            facts.containsKey(GCP_INSTANCE_ID.getValue()) || facts.containsKey(GCP_LICENSE_CODES.getValue());
+
+        if ((hasAWS && hasAzure) || (hasAWS && hasGCP) || (hasAzure && hasGCP)) {
+            throw new IllegalArgumentException("Multiple cloud providers facts were found.");
+        }
+        if (hasAWS) {
+            return "AWS";
+        }
+        if (hasAzure) {
+            return "AZURE";
+        }
+        if (hasGCP) {
+            return "GCP";
+        }
```

Comment:
It may make sense to make these cloud provider short name strings static and public variables on this class.

Reply:
done